UKHomeOffice/sgmr-service
# COP-3190+2768: Mark vessel fields as optional and add phone number validation for registering user

## Description
This branch marks the fields on the create a vessel page as optional and adds basic validation to the mobile number when a user registers (allows numbers and whitespaces only). 

To test:
- When signed out, attempt to register a user with a phone number that contains anything other than a digit or whitespace. Submit and you will see an error message. 
- Navigate to the create a vessel page (http://localhost:8080/vessels/save-vessel?source=vessels) - you will see optional on the relevant field labels. 



## Developer Checklist

\* Required






Ensure you delete the branch once the Pull Request is merged.
